### PR TITLE
Functional improvement.

### PR DESCRIPTION
**Pull request by andrei-maroz**
Added script for creating archive.
Added Jointoken and CustomData fields and logic for its.
Added Serial Console activation.

### CODE REVIEW

**Review by dskrypko — APPROVED** [review]

File: main/createUiDefinition.json [hunk 1]
```diff
@@ -34,18 +34,75 @@
         },
         "visible": true
       },
+      {
+        "name": "vmSize",
+        "type": "Microsoft.Compute.SizeSelector",
+        "label": "VM size",
+        "toolTip": "Select the size of VM.",
+        "recommendedSizes": [
+          "Standard_F4s"
```
Comment (joytandon17) [inline main/createUiDefinition.json:43]: I think recommended size needs to be "Standard_F8s"

File: main/createUiDefinition.json [hunk 2]
```diff
@@ -34,18 +34,75 @@
...
+              "label": "Custom Data",
+              "value": "Custom Data"
+            }
+          ],
+          "required": true
+        },
+        "visible": true
+      },
+      {
+        "name": "jointoken",
```
Comment (jkraj) [inline main/createUiDefinition.json:83]: will both jointoken and custom-data will be visible to user?

File: main/mainTemplate.json [hunk 3]
```diff
@@ -70,7 +117,7 @@
         "description": "Virtual network address prefix."
```
Comment (joytandon17) [inline main/mainTemplate.json:117]: We need to change the default vale of Virtual Network to 10.0.0.0/16, as maximum number of Private IP Address per Virtual Network can be 65536 and this should resolve the warning "Selected subscription doesn't have have enough available address space that meets all the constraints of the default virtual network specified." which we get as of now while creating deployment through offer.

File: main/createUiDefinition.json (hunk 2 above)
Comment (nnallamo) [reply main/createUiDefinition.json:83]: we have to hide this in final release . it is only for QA validation

**Review by nnallamo — APPROVED** [review]

File: main/createUiDefinition.json [hunk 4]
```diff
@@ -34,18 +34,75 @@
...
+          "validationMessage": "Insert validate Jointoken."
+        },
+        "visible": "[equals(basics('injecting'), 'Jointoken')]"
+      },
       {
         "name": "customData",
         "type": "Microsoft.Common.TextBox",
         "label": "Custom data",
-        "toolTip": "Pass custom data configuration.",
+        "toolTip": "In case if this field is filled, Jointoken field will be ignored.",
```
[the remark this answers is not in the document]
Comment (andrei-maroz) [reply main/createUiDefinition.json]: Changed toolTip to another one.

File: main/createUiDefinition.json (hunk 1 above)
Comment (andrei-maroz) [reply main/createUiDefinition.json]: Done

File: main/mainTemplate.json (hunk 3 above)
Comment (andrei-maroz) [reply main/mainTemplate.json:133]: A customer may have a different subscription.

**Review by dskrypko — APPROVED** [review]

File: main/createUiDefinition.json [hunk 5]
```diff
@@ -132,23 +164,68 @@
...
+            "toolTip": "Choose storage account for Serial console.",
+            "defaultValue": {
+              "type": "Standard_LRS"
+            },
+            "allowedValues": [
+              "Standard_LRS",
+              "Standard_GRS",
+              "Premium_LRS",
+              "Premium_ZRS"
+            ],
```
[the remark this answers is not in the document]
Comment (andrei-maroz) [reply main/createUiDefinition.json]: I did a few tests and I found that Serail console functionality works only with Standard_GRS storage account.

**Review by joytandon17 — APPROVED** [review]

**Pull request merged**